tensorflow/core/framework/device_factory.cc

```diff
@@ -63,17 +64,32 @@ int32 DeviceFactory::DevicePriority(const string& device_type) {
 }
 
 // static
-void DeviceFactory::Register(const string& device_type, DeviceFactory* factory,
-                             int priority) {
+// If subdevice type is not found in the device factories, that means
+// device is not registered. In this case, device type will be used as
+// subdevice type.
+string DeviceFactory::SubDeviceType(const string& device_type) {
+  tf_shared_lock l(*get_device_factory_lock());
+  std::unordered_map<string, FactoryItem>& factories = device_factories();
+  auto iter = factories.find(device_type);
+  if (iter != factories.end()) {
+    return iter->second.subdevice_type;
+  }
+  return device_type;
+}
+
+// static
```

Comment:
in original device_factory.cc, those functions declared in device_factory.h as static function are comment as static in the device_factroy.cc([DeviceFactory::DevicePriority](https://github.com/tensorflow/tensorflow/blob/06abb2cbb6d5902688bb291f080173446b8581fb/tensorflow/core/framework/device_factory.cc#L53), ), since I added a new static function before DeviceFactory::Register(), so I need to add this comment back.

```
// static
int32 DeviceFactory::DevicePriority(const string& device_type) {
  tf_shared_lock l(*get_device_factory_lock());
  std::unordered_map<string, FactoryItem>& factories = device_factories();
  auto iter = factories.find(device_type);
  if (iter != factories.end()) {
    return iter->second.priority;
  }

  return -1;
}

// static
void DeviceFactory::Register(const string& device_type, DeviceFactory* factory,
                             int priority) {
```

Reply:
Thanks for explaining!